### DISCUSSION TITLE
Outside requirements

### DISCUSSION THREAD

**Discussion by adam-fowler** [swiftlang/vscode-swift#406] [General]
# Outside requirements

This is a list of issues added to other projects that would improve the VS Code Swift extension. I will try to keep this up to date as new requirements/bugs appear or are resolved.

## Features

- Custom LSP request to return all tests and their location https://github.com/apple/sourcekit-lsp/issues/611
- Run formatter on CodeAction text. It is assumed by sourcekitd that `CodeAction` text will be formatted by another process. SourceKit-LSP should be doing this formatting. https://github.com/apple/sourcekit-lsp/issues/496
- To support Rename Symbol in VS Code we need to support LSP RenameRequest https://github.com/apple/sourcekit-lsp/issues/498
- SourceKit-LSP not aware of changes from other files. This is difficult to resolve but can be mitigated with LSP `textDocument/diagnostic` support. https://github.com/apple/sourcekit-lsp/issues/500
- Jump to definition for stdlib/Foundation. This is a very involved task which could result in a fair amount of work from both extension and SourceKit-LSP https://github.com/apple/sourcekit-lsp/issues/618
- Add custom snippet folder to package describe, so VS Code extension can catch new snippets without having to keep running `swift package describe` https://github.com/apple/swift-package-manager/issues/5750
- Add cross compilation arguments to source kit-lsp https://github.com/apple/sourcekit-lsp/issues/601

## Bugs

- Reword confusing error https://github.com/apple/sourcekit-lsp/issues/503
- Cannot support sanitizers while this occurs. Hang when swapping sanitizers https://github.com/apple/swift-package-manager/issues/4339
- Cannot skip tests when listing tests (Swift 5.8) https://github.com/apple/swift-package-manager/issues/5739
- Listing tests outputs error message (Swift 5.8) https://github.com/apple/swift-package-manager/issues/5744
- LLDB Python scripts are broken on Linux. This can break the finding of the Swift version of LLDB. https://github.com/apple/llvm-project/issues/4770
- Jump to definition requires reload after build https://github.com/apple/sourcekit-lsp/issues/620
- `workspace-state.json` isn't updated when dependency removed. If it was we could use it as a source of truth for project dependencies https://github.com/apple/swift-package-manager/issues/5727

**Comment by stevapple** [comment]
Perhaps give apple/sourcekit-lsp#601 a shot as it’s one of the main blockers for #386. It’s hard to pass all the configurations to SK-LSP, but much easier if we can use destination file directly.

Also wants to mention how CLion deals with corelib/stdlib jump-to-definitions: it preloads all the available modules, generates `.swift` files with public interface and documentation, and caches them to disk. This is, however, _very very time-consuming_, but we can see if it’s possible to do it for a single module instantly (to achieve an Xcode-like UX).

**Reply by adam-fowler** [reply]
Added https://github.com/apple/sourcekit-lsp/issues/601

**Reply by adam-fowler** [reply]
I added https://github.com/swift-server/vscode-swift/issues/408 for generating module interface files
